pkg/domain/podman/credential_handler.go

```diff
@@ -0,0 +1,280 @@
+	"github.com/skupperproject/skupper/client/generated/libpod/client/volumes"
+	"github.com/skupperproject/skupper/client/podman"
+	"github.com/skupperproject/skupper/pkg/certs"
+	"github.com/skupperproject/skupper/pkg/kube"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type PodmanCredentialHandler struct {
+	cli *podman.PodmanRestClient
+}
+
+func (p *PodmanCredentialHandler) ListCertAuthorities() ([]types.CertAuthority, error) {
+	list, err := p.cli.VolumeList()
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving certificate authorities - %w", err)
+	}
+	cas := []types.CertAuthority{}
+	for _, v := range list {
+		if v.Labels == nil {
+			continue
+		}
+		if kind, ok := v.Labels[types.SkupperTypeQualifier]; ok && kind == "CertAuthority" {
+			cas = append(cas, types.CertAuthority{Name: v.Name})
+		}
+	}
+	return cas, nil
+}
+
+func (p *PodmanCredentialHandler) ListCredentials() ([]types.Credential, error) {
+	list, err := p.cli.VolumeList()
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving certificate authorities - %w", err)
+	}
+	creds := []types.Credential{}
+	for _, v := range list {
+		if v.Labels == nil {
+			continue
+		}
+		// ignoring volumes that are not credentials
+		if kind, ok := v.Labels[types.SkupperTypeQualifier]; !ok || kind != "Credential" {
+			continue
+		}
+		files, err := v.ListFiles()
+		if err != nil {
+			return nil, fmt.Errorf("error listing credential content - %w", err)
+		}
+		cred := types.Credential{
+			Name: v.Name,
+		}
+
+		// Validating if CA is provided
+		empty := false
+		for _, file := range files {
+			// CA defined
+			if file.Name() == types.ClaimCaCertDataKey {
+				var ca *types.CertAuthority
+				content, err := v.ReadFile(file.Name())
+				if err != nil {
+					return nil, fmt.Errorf("error validating cert authority - %w", err)
+				}
+				ca = p.getCertAuthorityForCaCrt(content)
+				if ca != nil {
+					cred.CA = ca.Name
+				}
+			} else if file.Name() == "connect.json" {
+				cred.ConnectJson = true
+			} else if file.Name() == "tls.crt" {
+				dataStr, err := v.ReadFile(file.Name())
+				if dataStr == "" {
+					empty = true
+					continue
+				}
+				if err != nil {
+					return nil, fmt.Errorf("error reading tls.crt file from volume %s - %w", v.Name, err)
+				}
+				cn, hostnames, err := getTlsCrtHostnames([]byte(dataStr))
+				if err != nil {
+					return nil, fmt.Errorf("unable to retrieve subject and hostnames from tls.crt under %s - %w", v.Name, err)
+				}
+				cred.Subject = cn
+				if len(hostnames) > 0 {
+					cred.Hosts = hostnames
+				} else {
+					cred.Simple = true
+				}
+			}
+		}
+
+		if !empty {
+			creds = append(creds, cred)
+		}
+	}
+	return creds, nil
+}
+
+func NewPodmanCredentialHandler(cli *podman.PodmanRestClient) *PodmanCredentialHandler {
+	return &PodmanCredentialHandler{
+		cli: cli,
+	}
+}
+
+func (p *PodmanCredentialHandler) LoadVolumeAsSecret(vol *container.Volume) (*corev1.Secret, error) {
```

Comment:
does this need to be on the handler?

Reply:
This is just part of the podman.CredentialHandler implementation.
The `CredentialHandler` interface does not define this method.